apps/login/src/lib/service.ts

```diff
@@ -108,3 +109,20 @@ export function getServiceUrlFromHeaders(headers: ReadonlyHeaders): {
     serviceUrl: instanceUrl,
   };
 }
+
+export function constructUrl(request: NextRequest, path: string) {
+  const forwardedProto = request.headers.get("x-forwarded-proto") ?? "https";
+
+  const forwardedHost =
+    request.headers.get("x-zitadel-forward-host") ??
+    request.headers.get("host");
+  const basePath = process.env.NEXT_PUBLIC_BASE_PATH || "";
+  return new URL(
+    `${basePath}${path}`,
+
+    // keep this check to allow localhost for local development
+    forwardedHost?.startsWith("http")
+      ? forwardedHost
+      : `${forwardedProto}://${forwardedHost}`,
+  );
```

Comment:
forwardedHost never starts with http

```suggestion
  return new URL(
    `${basePath}${path}`,
    `${forwardedProto}//${forwardedHost}`,
  );
```

Reply:
are you sure.. this could be localhost:3000 as it is also falling back to the host header. so this check is still valid.

Reply:
huh? `localhost:3000` doesn't start with `http`.

Reply:
I think the port is already with a semicolon in the forwardedHost if non-standard